x/oracle/client/cli/txRegisterOracle.go

```diff
@@ -16,39 +16,70 @@ import (
 
 func CmdRegisterOracle() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "register-oracle [unique ID] [node public key] [node public key remote report] [trusted block height] [trusted block hash] [endpoint] [oracle's commission rate]",
+		Use:   "register-oracle",
 		Short: "Register a new oracle",
-		Args:  cobra.ExactArgs(7),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			clientCtx, err := client.GetClientTxContext(cmd)
 			if err != nil {
 				return err
 			}
 
 			oracleAddress := clientCtx.GetFromAddress().String()
-			nodePubKey, err := base64.StdEncoding.DecodeString(args[1])
+
+			uniqueID, err := cmd.Flags().GetString(flagOracleUniqueID)
+			if err != nil {
+				return fmt.Errorf("failed to get oracle unique ID")
+			}
+
+			nodePubkeyStr, err := cmd.Flags().GetString(flagNodePublicKey)
+			if err != nil {
+				return fmt.Errorf("failed to get node public key")
+			}
+
+			nodePubKey, err := base64.StdEncoding.DecodeString(nodePubkeyStr)
 			if err != nil {
 				return err
 			}
 
-			nodePubKeyRemoteReport, err := base64.StdEncoding.DecodeString(args[2])
+			nodePubkeyRemoteReportStr, err := cmd.Flags().GetString(flagNodePubKeyRemoteReport)
+			if err != nil {
+				return fmt.Errorf("failed to get node public key remote report")
+			}
+
+			nodePubKeyRemoteReport, err := base64.StdEncoding.DecodeString(nodePubkeyRemoteReportStr)
 			if err != nil {
 				return err
 			}
 
-			trustedBlockHeight, err := strconv.ParseInt(args[3], 10, 64)
+			trustedBlockHeightStr, err := cmd.Flags().GetString(flagTrustedBlockHeight)
+			if err != nil {
+				return fmt.Errorf("failed to get trsuted block height")
+			}
+
+			trustedBlockHeight, err := strconv.ParseInt(trustedBlockHeightStr, 10, 64)
 			if err != nil {
 				return err
 			}
 
-			trustedBlockHash, err := hex.DecodeString(args[4])
+			trustedBlockHashStr, err := cmd.Flags().GetString(flagTrustedBlockHash)
+			if err != nil {
+				return fmt.Errorf("failed to get trsuted block hash")
+			}
+
+			trustedBlockHash, err := hex.DecodeString(trustedBlockHashStr)
 			if err != nil {
 				return err
 			}
 
-			endpoint := args[5]
+			endpoint, err := cmd.Flags().GetString(flagOracleEndpoint)
+			if err != nil {
+				return fmt.Errorf("failed to get oralce end point")
+			}
 
-			oracleCommissionRateStr := args[6]
+			oracleCommissionRateStr, err := cmd.Flags().GetString(flagOracleCommRate)
+			if err != nil {
+				return fmt.Errorf("failed to get oralce commission rate")
+			}
 
 			if len(oracleCommissionRateStr) == 0 {
```

Comment:
There is no need to check the length of commission rate because `sdk.NewDecFromStr` will do.

Reply:
Oh, Thanks! I fixed it!